### PR TITLE
More fixes to payments tests

### PR DESCRIPTION
**Pull request by marcoscaceres**
These tests were buggy... but now they are not :) 

Safari is happy...

### CODE REVIEW

File: payment-request/payment-request-show-method.https.html [hunk 1]
```diff
@@ -51,17 +51,16 @@
...
-    "test: throws if the promise [[state]] is not 'created'"
-  );
+  await promise_rejects_dom(t, "SecurityError", request.show());
+
+  await test_driver.bless();
   const acceptPromise = request.show(); // Sets state to "interactive"
+
   // No user activation...
-  await promise_rejects_dom(t, "SecurityError", request.show());
+  await promise_rejects_dom(t, "InvalidStateError", request.show());
```
Comment (romandev) [inline payment-request/payment-request-show-method.https.html:60]: This `InvalidStateError` doesn't seem to be related to user activation. Am I missing something?

Comment (marcoscaceres) [reply payment-request/payment-request-show-method.https.html:60]: Yeah, that comment needs to go...

File: payment-request/payment-request-show-method.https.html [hunk 2]
```diff
@@ -51,17 +51,16 @@
...
   const request = new PaymentRequest(defaultMethods, defaultDetails);
-  await test_driver.bless(
-    "test: throws if the promise [[state]] is not 'created'"
-  );
+  await promise_rejects_dom(t, "SecurityError", request.show());
+
+  await test_driver.bless();
   const acceptPromise = request.show(); // Sets state to "interactive"
+
   // No user activation...
```
Comment (marcoscaceres) [inline payment-request/payment-request-show-method.https.html:59]:
```suggestion
```

**Comment by marcoscaceres** [comment]
@romandev, would you mind reviewing these?

**Review by romandev — APPROVED** [review]
lgtm
